grid/chat/channels/maps/discord.suma

```diff
@@ -0,0 +1,183 @@
+profile = "chat/channels@1.0"
+provider = "discord"
+
+"""
+GetChannels map
+"""
+map GetChannels {
+  return map error if (input.server === undefined) {
+    title = "Input parameter `server` is required in case of discord provider"
+  }
+
+  http GET "/guilds/{input.server}/channels" {
+    request "application/json" {
+      headers {
+        "Authorization" = `Bot ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      channels = call MapChannels(channels = body)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+
+      map result {
+        channels = channels
+        rateLimit = rateLimit
+      }
+    }
+
+    response {
+      error = call MapDiscordError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapChannels
+Returns array mapping channels in received body
+"""
+operation MapChannels {
+  channels = []
+
+  channels = args.channels.filter(channel => [0, 1, 3, 5].includes(channel.type)).map(channel => {
+    const discordEpoch = 1420070400000;
+    const getTimestamp = (snowflake) => Number(BigInt(snowflake) >> BigInt(22)) + discordEpoch;
```

Comment:
Can you construct that outside of the map function?

```
channels = []
getTimestamp = (snowflake) => Number(BigInt(snowflake) >> BigInt(22)) + 1420070400000;
```

Also it would be helpful to add a comment with link to Discord docs for this algorithm

Reply:
Not possible 😢  I'll comment it more